holoviews/plotting/bokeh/raster.py

```diff
@@ -161,6 +164,239 @@ def get_data(self, element, ranges, style):
+
+        return array[:, 0], array[:, 1]
+
+
+    @staticmethod
+    def _extract_labels(mapper):
+        """Extracts text label and radiant for segment texts.
+
+        """
+
+        values = [(text, ((end - start) / 2) + start)
+                  for text, (start, end) in mapper.items()]
+
+        text, radiants = zip(*values)
+
+        return text, np.array(radiants)
+
+
+    @staticmethod
+    def _compute_separations(inner, outer, angles):
+        """Compute x and y positions for separation lines for given angles.
+
+        """
+
+        y_start = np.sin(angles) * inner + 0.5
+        y_end = np.sin(angles) * outer + 0.5
+
+        x_start = np.cos(angles) * inner + 0.5
+        x_end = np.cos(angles) * outer + 0.5
+
+        return zip(x_start, x_end), zip(y_start, y_end)
+
+
+    def _postprocess_hover(self, renderer, source):
+        """Limit hover tool to annular wedges only.
+
+        """
+
+        if isinstance(renderer.glyph, AnnularWedge):
+            super(RadialHeatMapPlot, self)._postprocess_hover(renderer, source)
+
+
+    def get_extents(self, view, ranges):
+        lower = -self.padding_outer
+        upper = 1 + self.padding_outer
+        return (lower, lower, upper, upper)
+
+
+    def get_data(self, element, ranges, style):
+
+        # dimension labels
+        dim_labels = element.dimensions(label=True)[:3]
+        x, y, z = [dimension_sanitizer(d) for d in dim_labels]
+
+        if self.invert_axes: x, y = y, x
+        cmapper = self._get_colormapper(element.vdims[0], element, ranges, style)
+
+        # define default CDS data mappings
+        map_annular = dict(x=0.5, y=0.5,
+                           inner_radius="inner_radius",
+                           outer_radius="outer_radius",
+                           start_angle="start_angle",
+                           end_angle="end_angle",
+                           fill_color={'field': z, 'transform': cmapper})
+
+        map_text = dict(x="x", y="y", text="text",
+                        angle="angle", text_align="center")
+
+        map_multi_line = dict(xs="xs", ys="ys")
+
+        mapping = {'annular_wedge_1': map_annular,
+                   'text_1': map_text,
+                   'multi_line_1': map_multi_line}
+
+        if self.static_source:
+            return {}, mapping, style
+
+        # get raw values
+        aggregate = element.gridded
+        xvals = aggregate.dimension_values(x)
+        yvals = aggregate.dimension_values(y)
+        zvals = aggregate.dimension_values(2, flat=True)
+
+        # pretty print x and y dimension values if necessary
+        def _pprint(dim_label, vals):
+            if vals.dtype.kind not in 'SU':
+                dim = aggregate.get_dimension(dim_label)
+                return [dim.pprint_value(v) for v in vals]
+
+            return vals
+
+        xvals = _pprint(x, xvals)
+        yvals = _pprint(y, yvals)
+
+
+        # get orders
+        order_segment = self._extract_implicit_order(xvals)
```

Comment:
I'd actually just use ``aggregate.dimension_values(x, expanded=False)`` and ``aggregate.dimension_values(y, expanded=False)``  here instead.

Reply:
Agreed - definitely better.